### DISCUSSION TITLE
Questions on Durations for Lot and Contract To Be Checked

### DISCUSSION THREAD

**Discussion by giorgialodi** [eprocurementontology/eprocurementontology#515] [Ideas]
I was analysing the OWL restriction file  (version 2.0.1) for opening possible issues in case of strange things noticed.

My attention was caught by the property hasContractDuration between the Lot and the Period.  The definition is "The estimated date when the contract is to be started."

Question: why is it called hasContractDuration if it is a property for the Lot? Shouldn't be for the Contract a property called hasContractDuration? 
In the name of the property the fact that is estimated is not captured. In addition it is a date, not a period, according to the definition.
If we stick to the definition, something like hasContractEstimatedStartingDate seems more inline (which i s a xsd:date not a period).
What do you think about it?

**Comment by SellittoGiampaolo** [comment]
A Lot usually generates a contract, which in turn has an expected duration.
The contract duration must be linked to the specific lot, since you can have lots with different duration and different starting dates
These properties are already known before the call for tender and are part of the information that must be published, so they are properly comprised in the information known before the contract is signed - the piece of information regarding the contract is filled at a later date, and it would not be possible to include the length of the contract when you do not have the core information in place.
I think that the modeling is correct.

**Comment by ec-mcs** [comment]
How would it be for a DPS where you naturally have many contracts per lot? Or how about mini-tenders for FA?

**Comment by giorgialodi** [comment]
Sorry, probably I was not able to express myself correctly. I am not suggesting to remove the property. I am just suggesting that a better naming can be possible, that can represent what is included in the current definition for that property; that is, the fact that it is an **estimated** starting date.
For me  "hasContractDuration" recalls the duration (start date and end date) of a contract (signed).  It was anyway a question :)

**Comment by SellittoGiampaolo** [comment]
@ec-mcs In the DPS we track:
- the call for tender that establishes a DPS - there you know the duration and the estimated starting date of the DPS 
- the calls for tender in the DPS, that in turn give life to contracts that have their own length and starting date, possibly different from the DPS duration.
This is the main reason you need to have the length at the time you publish the notice, before the contract is signed - with the DPS the duration is established also in absence of the adhesion from  economic operators and an EO can enroll and have a contract inside the DPS only during its duration, not after.

**Comment by paulakeen** [comment]
The properties between Contract, Lot and Period need to be reviewed since the predicates do not make sense in some cases. For example, in "Contract hasEvaluationDuration Period", and other... (contracts are not evaluated, but lots are). Other issues with these properties may also exist.

**Comment by andreea-pasare** [comment]
At the moment, there is a relation from Contract to Duration, epo:hasEstimatedDuration:
![image](https://user-images.githubusercontent.com/88081600/203766118-0892c9d2-6130-4e04-8d28-e80e7c7ed43e.png)

The definition for this relations was changed to `Relation indicating a Contract has an estimated Duration.`
This is related to #416 and will be further investigated in the near future.

**Comment by muricna** [comment]
If the definication is indicating a Contract has an estimated Duration the reuse of this predicate is not possible.

**Comment by andreea-pasare** [comment]
@muricna Should we also change the name of the predicate to epo:hasContractEstimatedDuration? 
Also, do we want this relation moved at the level of the ContractTerm since we have the following relations from ContractTerm:
![image](https://user-images.githubusercontent.com/88081600/203772584-2427a668-e11d-4d78-a779-5203819a380d.png)

**Comment by muricna** [comment]
If it is at ContractTerm level I am not sure it is need in the contract
